box-share-sdk/src/main/java/com/box/androidsdk/share/utils/SharedLinkAccessToggleListeners.java

```diff
@@ -0,0 +1,40 @@
+package com.box.androidsdk.share.utils;
+
+import com.box.androidsdk.content.models.BoxBookmark;
+import com.box.androidsdk.content.models.BoxItem;
+import com.box.androidsdk.content.models.BoxSharedLink;
+import com.box.androidsdk.share.usx.fragments.SharedLinkAccessFragment;
+
+public class SharedLinkAccessToggleListeners {
+
+    public static void onAccessLevelCheckChanged(boolean checked, BoxSharedLink.Access access, SharedLinkAccessFragment.SharedLinkAccessNotifiers notifiers) {
+        if (checked) {
+            notifiers.notifyAccessLevelChange(access);
+        }
+    }
+
+
+    public static void onDownloadToggle(boolean checked, BoxItem shareItem, SharedLinkAccessFragment.SharedLinkAccessNotifiers notifiers) {
+        if (shareItem instanceof BoxBookmark || shareItem.getSharedLink().getPermissions().getCanDownload() == checked ){
```

Comment:
Same question..needs null checks, or are these always present? sharedlink, permissions

Reply:
User cannot get to this screen if shared link was not present in the first place. I believe permissions always exist though.

Reply:
I can do a check at a central place to make it safer though (for the shared link). I will be doing that just in case.